src/repr/src/strconv.rs

```diff
@@ -127,18 +126,7 @@ where
 
 /// Parses an `f32` from `s`.
 pub fn parse_float32(s: &str) -> Result<f32, ParseError> {
-    let s = UncasedStr::new(s.trim());
-    if s == "inf" || s == "infinity" || s == "+inf" || s == "+infinity" {
-        Ok(f32::INFINITY)
-    } else if s == "-inf" || s == "-infinity" {
-        Ok(f32::NEG_INFINITY)
-    } else if s == "nan" {
-        Ok(f32::NAN)
-    } else {
-        s.as_str()
-            .parse()
-            .map_err(|e| ParseError::new("real", s.as_str()).with_details(e))
-    }
+    fast_float::parse(s).map_err(|e| ParseError::new("real", s).with_details(e))
```

Comment:
Let's ditch the `with_details` bit here and below. The error returned from `fast_float` is always "error while parsing a float" https://github.com/aldanor/fast-float-rust/blob/5fa4f295e742e4882eb9efabe7778f45d450cb1b/src/lib.rs#L64

which is redundant with our own message in `ParseError`!

Reply:
Do you expect that it will always return `error while parsing float`? I'm not sure I see the harm in maintaining `with_details` given that they may change their error handling in the future?

Reply:
> Do you expect that it will always return `error while parsing float`? I'm not sure I see the harm in maintaining `with_details` given that they may change their error handling in the future?

The top priority here should be making Materialize return good errors with the code as it exists today. The problem is that the current display makes for a confusing error message:

```
invalid input syntax for real: error while parsing a float: "dog"
```

Removing the `with_details` call yields

```
invalid input syntax for real:  "dog"
```

which is much clearer.

Separately, you don't get fast at parsing floats by providing good error messages. I would be pretty surprised if they manage to meaningfully improve error messages in this library.

Reply:
That seems fair given the use case!